src/main/webapp/app/exam/manage/exams/exam-exercise-import/exam-exercise-import.component.ts

```diff
@@ -192,7 +219,10 @@ export class ExamExerciseImportComponent implements OnInit {
     validateTitleOfProgrammingExercise(exercise: Exercise): boolean {
         return (
             // eslint-disable-next-line @typescript-eslint/no-non-null-asserted-optional-chain
-            exercise.title?.length! > 0 && this.titleNamePattern.test(exercise.title!) && exercise.title !== this.getBlocklistTitleOfProgrammingExercise(exercise.id!)
+            exercise.title?.length! > 0 &&
+            this.titleNamePattern.test(exercise.title!) &&
+            !this.exercisesWithDuplicatedTitles.has(exercise.id!) &&
+            (exercise.title !== this.getBlocklistTitleOfProgrammingExercise(exercise.id!) || this.getBlocklistShortNameOfProgrammingExercise(exercise.id!) === '')
```

Comment:
```suggestion
            exercise.title !== this.getBlocklistTitleOfProgrammingExercise(exercise.id!)
```
When the second condition is checked, you already know that the result from that function call is equal to `exercise.title`. So, you are effectively checking `exercise.title === ''`. This is never true, because otherwise, the line `exercise.title?.length! > 0` would have short-circuited.

Reply:
This additional check for `this.getBlocklistShortNameOfProgrammingExercise(exercise.id!) === ''` is for the case a programming exercise with the same project name already exists on bamboo. In this case the exercise name is not valid and should not be allowed. 

Before, `this.titleAndShortNameOfProgrammingExercises` was only used as the placeholder in this case. Now however it is also used as the placeholder in case there are already duplicated titles when selecting the exam to import (look at the video in the PR description, I think it explains it the best :sweat_smile: ). 
Just because it was a duplicate once at the start, this one programming exercise should still be alowed to have the same title (that was once considered a duplicate) becouse it is now possibly no longer one (e.g. due to renaming the other title).

Reply:
Ah, I just misread the function names to be the same, but they're not 🤦